### DISCUSSION TITLE
Free or expand space on the Onion system root drive

### DISCUSSION THREAD

**Discussion by edbadaaz** [Security-Onion-Solutions/securityonion#9885] [General]
Good afternoon, I am with the Onion root drive, showing full, my onion is installed with Centos7, I have already performed the expansion of the disk in the VM but when performing the command for expansion, nothing is done.

![image](https://user-images.githubusercontent.com/126823982/223553424-43b10cc2-e763-4808-a219-8cc85422a395.png)

Is there any way to expand or remove unnecessary files?

**Comment by cm-ops** [comment]
What command are you running?

**Reply by edbadaaz** [reply]
Hi this comand:   lvresize --size +10G /dev/mapper/system-root

**Reply by cm-ops** [reply]
Have you tried `lvextend` instead of `lvresize`?

**Reply by edbadaaz** [reply]
Yes, but notworking

![image](https://user-images.githubusercontent.com/126823982/224380175-8ce7711d-fe24-4042-bd92-96269fa47ba0.png)

![image](https://user-images.githubusercontent.com/126823982/224380149-3b6938d9-e927-4cf2-90da-872ca34234ab.png)

I tried sending the resize2fs command and still no success. It generates the error below:

![Uploading image.png…]()

**Comment by cm-ops** [comment]
Have you tried `xfs_growfs` command instead of `resize2fs`?

**Reply by edbadaaz** [reply]
Yes, but not working...

![image](https://user-images.githubusercontent.com/126823982/225001543-b5bb90c7-d752-41db-8ba6-8135ecd53b1d.png)

**Reply by edbadaaz** [reply]
It gives me this result, but does no action @cm-ops

**Comment by cm-ops** [comment]
Okay, let's back up a bit. You expanded the disk in the VM, what other steps were taken before trying to expand the fs?

**Reply by edbadaaz** [reply]
Not at first, I just performed the disk expansion in the vmware hypervisor. And with the commands we talked about above, I tried to expand the partition.
I managed to add the space to the disk on the system using parted, but I can't expand the partition.

@cm-ops

**Reply by cm-ops** [reply]
What does `pvdisplay` show? Is there free space for the PV?

**Reply by edbadaaz** [reply]
@cm-ops 
![image](https://user-images.githubusercontent.com/126823982/225930687-516d34e4-565c-4208-87b0-a58c61e5a28b.png)

No space for the pv... that's what perplexes me.

And the disk is expanded correctly.

**Reply by isaacgolding** [reply]
So the problem is very likely that since the drive is at 100% full you don't have the physical space on the drive to expand the drive itself (while it is running).   More than likely the only fix is to boot the system from some type of Linux live CD distro, then you can use it's shell or GUI tools to expand the filesystem.     More importantly from what I can see of your partition table in the screencap it looks like you aren't even meeting the MIN requirements for a standalone install.   

If this is a test/lab system I would suggest going back and meeting the min requirements for drive space.   And even at min specs the log files will start filling up partitions really really fast.  

https://docs.securityonion.net/en/2.3/hardware.html?highlight=min%20specs

**Comment by edbadaaz** [comment]
I understand that you don't have the minimum required for a standalone laboratory installation. 
My hardware resources for the lab are limited. That is why I am here to ask for help. As I have already said, I have expanded the virtual disk and still cannot expand the partition.
